BackgammonPlay.py

```diff
@@ -0,0 +1,256 @@
+import random
+
+
+class InvalidMoveError(Exception):
+    pass
+
+
+class Board(object):
+    def __init__(self):
+        self.board = [[] for i in range(24)]
+
+    def get_board(self):
+        return self.board
+
+    def start_board(self, sign1, sign2):
+        self.board[0] = 2 * [sign1]
+        self.board[5] = 5 * [sign2]
+        self.board[7] = 3 * [sign2]
+        self.board[11] = 5 * [sign1]
+        self.board[12] = 5 * [sign2]
+        self.board[16] = 3 * [sign1]
+        self.board[18] = 5 * [sign1]
+        self.board[23] = 2 * [sign2]
+
+    def change_board(self, index_now, steps, sign):
+        try:
+            if (sign in self.board[index_now] and (
+                    self.board[index_now + steps] == [] or sign in (self.board[index_now + steps]))):
+                self.board[index_now] = self.board[index_now][:-1]
+                self.board[index_now + steps].append(sign)
+                return
+            else:
+                return InvalidMoveError('invalid move')
+        except Exception as e:
+            return e
+
+    def change_board_level_2(self, index):
+        # get valid input
+        self.board[index] = self.board[index][:-1]
+
+
+class Player(object):
+    def __init__(self, sign, name):
+        self.sign = sign
+        self.name = name
+
+    def get_sign(self):
+        return self.sign
+
+    def get_name(self):
+        return self.name
+
+
+class Game(object):
+    def __init__(self, player1, player2, board):
+        self.p1 = player1
+        self.p2 = player2
+        self.board = board
+        self.count = -1
+        self.val_cube = (0, 0)
+
+    def player_now(self):
+        self.count = self.count + 1
+        if self.count % 2 == 0:
+            return self.p1
+        return self.p2
+
+    def rolling_the_cubes(self):
+        num1 = random.randint(1, 6)
+        num2 = random.randint(1, 6)
+        self.val_cube = num1, num2
+        return self.val_cube
+
+
+class SpecificField(Game):
+    pass
+
+    def print_board(self):
+        print_str_board(self.board.get_board())
+
+    def turn_in_game(self, player):
```

Comment:
this is a game functionality - dependancy invertion!
make the game functionality be in the game class and use the field to connect to the players